### PR TITLE
ECIP 1096: 51% Attack protection system based on Bitcoin Merged Mining

### PR DESCRIPTION
**Pull request by martinmedina**
**Draft.** [51% Attack protection system based on Bitcoin Merged Mining](https://github.com/martinmedina/ECIPs/blob/patch-1/_specs/ecip-51-attack-protection-system.md) 
**Discussion.** https://github.com/ethereumclassic/ECIPs/issues/343

### CODE REVIEW

File: _specs/ecip-51-attack-protection-system.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+### Inclusion of VisibilityProof Proofs
+
+On a first version, EtHash miners will be the only ones that will have the possibility to add VisibilityProofs to their blocks.
+
+In the future, the ETC network can propagate VisibilityProofs messages and we can let any user participate and collect fees from this system. This requires no changes to consensus. 
+
+### Rewards
+
+If a miner decides to add a VisibilityProof message in a slot, it will be rewarded by an extra subsidy. To prevent changing the ETC supply function, the extra subsidy will be obtained by diluting all remaining block subsidies by a fixed percentage. The exact amount is to be decided by the community but we propose this to be 10% of the current subsidy, starting at 1% and increasing until 10% during a period of 3 months. This will allow for EtHash miners to adapt and provides enough incentives for them to start using the RSK UMM system to obtain VisibilityProofs for their blocks.
```
Comment (q9f) [inline _specs/ecip-51-attack-protection-system.md:114]: What if a miner does not include a proof, would that reward be lost or burnt?

File: _specs/ecip-51-attack-protection-system.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+The spirit of this proposal is to present a simple way to provide ETC with an otional system that can temporarily or permanently protect it from 51% attacks by establishing a strong deterrent. To achieve that:
+
+* VisibilityProof messages are included as uncles so the system is the least invasive way as possible.
+* In the current version, VisibilityProof messages are included by mining nodes. However, the system allows any user to participate in the VisibilityProof system.  This extension requires adding a new message to the wire protocol the relay of VisibilityProof proofs. 
+* By having a non-mandatory addition of VisibilityProof messages, ETC blockchain can keep working normally even if with no VisibilityProof on the blocks. 
+
+## Universal Merged Mining
+
+Universal Merged Mining offers access to Bitcoin SHA-256 PoW by forwarding the to-be-validated information through smart-contracts deployed on the RSK blockchain.
```
Comment (q9f) [inline _specs/ecip-51-attack-protection-system.md:139]: I understand where you are coming from, could you elaborate why RSK and what feature is missing from Classic to leverage such a functionality on Classic directly?

**Review by q9f — COMMENTED** [review]
LGTM if you could add the number to the block header and rename the file to `ecip-1096.md`

**Comment by q9f** [comment]
Could you attend the public call later today to champion your ECIP? ref #333

File: _specs/ecip-51-attack-protection-system.md (hunk 2 above)
Comment (SergioDemianLerner) [reply _specs/ecip-51-attack-protection-system.md:139]: RSK has direct access to ~40% (sometimes >50%) to the Bitcoin miners. Smart contracts can communicate directly with Bitcoin miners, and Bitcoin miners can collect revenue directly from these smart-contracts. By using the auction contracts in RSK the ETC community leverages on this established technical and economic channel.

Comment (SergioDemianLerner) [reply _specs/ecip-51-attack-protection-system.md:139]: The doesn't mean that, in the future, ETC can deploy their own ETC nodes for Bitcoin mining pools. RSK is currently the most cost-efficient way to do it, and it's 100% operational.

File: _specs/ecip-51-attack-protection-system.md (hunk 1 above)
Comment (SergioDemianLerner) [reply _specs/ecip-51-attack-protection-system.md:114]: "lost" and "burnt" is the same. Maybe you mean accumulated (so the next miners gets a higher incentive) or burt. We (the authors) can define that, but I think this should be up to the ETC community to decide.

**Comment by SergioDemianLerner** [comment]
> Could you attend the public call later today to champion your ECIP? ref #333

We will. I've updated the ECIP with some new explanations about the relation and synergies between this ECIP and PirlGuard to make better use of the time in the call. See new section "Detecting and Preventing Hidden Double-Spend Attacks"

File: _specs/ecip-51-attack-protection-system.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+```
+1. **hit**:  hash=0000000000000000000017b25971a701, blockHeight=644735, parent =0000000000000000000d9e9adbe0d01
+2. **conf**: hash=0000000000000000000a34ee53a217b6, blockHeight=644736,  Parent=0000000000000000000017b25971a701
+3. **conf**: hash=0000000000000000000db9f0fb12c4e9, blockHeight=644737, Parent=0000000000000000000a34ee53a217b6
+4. **hit**:  hash=0000000000000000000a7470181b940a2, blockHeight=644740, Parent=0000000000000000000a34ee53a217b6
+5. **conf**: hash=000000000000000000025fdd6edcee338, blockHeight=644741, Parent=0000000000000000000a7470181b940a
```
Comment (gtklocker) [inline _specs/ecip-51-attack-protection-system.md:49]:
```suggestion
5. **conf**: hash=000000000000000000025fdd6edcee338, blockHeight=644741, Parent=0000000000000000000a7470181b940a2
```

**Review by gtklocker — CHANGES_REQUESTED** [review]

File: _specs/ecip-51-attack-protection-system.md (hunk 3 above)
Comment (martinmedina) [reply _specs/ecip-51-attack-protection-system.md]: Done, thanks for pointing that out!

File: _specs/ecip-51-attack-protection-system.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
+---
+lang: en
+ecip: ECIP-10XX
```
[the remark this answers is not in the document]
Comment (martinmedina) [reply _specs/ecip-51-attack-protection-system.md]: Done

**Review by realcodywburns — APPROVED** [review]
lgtm merge as draft

**Pull request merged**